1.8/development/plugins/creating-modifying-templates.md

```diff
@@ -94,3 +94,46 @@ global $user;
 
 $user['favorite_colour'] = 'Blue';
 ```
+
+## Caching Templates
+Don't forget to cache the template. This is important for pages with big load and prevent the database server from extra querys.
+
+### Caching a template
+An simple way to cache an template only at the place where it's needed, is to use the `THIS_SCRIPT` definition.
+For example, the template is needed in the postbit. Then we can load it in the showthreads.php.
+```php
+if(THIS_SCRIPT == 'showthread.php')
+{
+    global $templatelist;
+    if(isset($templatelist))
+    {
+        $templatelist .= ',';
+    }
+    $templatelist .= 'hello_world_template';
+}
+```
+Another example, here are some more templates needed on the member page. Then we can load it in the member.php.
+```php
+if(THIS_SCRIPT == 'member.php')
+{
+    global $templatelist;
+    if(isset($templatelist))
+    {
+        $templatelist .= ',';
+    }
+    $templatelist .= ',hello_world_template_second';
+    $templatelist .= ',hello_world_template_third';
+}
+```
+
+### Testing if the template get cached
+You can simply test if the templates get cached correctly or not. Open the page where they should get cached and add a `?debug&debug=1` or `&debug&debug=1` to the url into the browser.
+Now you see the `MyBB Debug Information` page. The first table called `Page Generation Statistics` contain a row with the template cache information. 
+
+#### There is an template that not get cached:
```

Comment:
Consider changing to "If a template does not get cached" or something and remove the colon. Headings don't need colons at the end of them.